enterprise_access/apps/events/utils.py

```diff
@@ -0,0 +1,147 @@
+            })
+
+        new_producer = SerializingProducer(producer_settings)
+        cls._type_to_producer[event_type] = new_producer
+        return new_producer
+
+
+def create_topic_if_not_exists(topic_name):
+    """
+    Create a topic in the event bus
+    :param topic_name: topic to create
+    """
+    KAFKA_ACCESS_CONF_BASE = {'bootstrap.servers': getattr(settings, 'KAFKA_BOOTSTRAP_SERVER', '')}
+
+    if settings.KAFKA_API_KEY and settings.KAFKA_API_SECRET:
+        KAFKA_ACCESS_CONF_BASE.update({
+            'sasl.mechanism': 'PLAIN',
+            'security.protocol': 'SASL_SSL',
+            'sasl.username': settings.KAFKA_API_KEY,
+            'sasl.password': settings.KAFKA_API_SECRET,
+        })
+
+    admin_client = AdminClient(KAFKA_ACCESS_CONF_BASE)
+
+    topic = NewTopic(topic_name,
+                                   num_partitions=settings.KAFKA_PARTITIONS_PER_TOPIC,
+                                   replication_factor=settings.KAFKA_REPLICATION_FACTOR_PER_TOPIC)
+    # Call create_topics to asynchronously create topic.
+    # Wait for each operation to finish.
+    topic_futures = admin_client.create_topics([topic])
+
+    # TODO: (ARCHBOM-2004) programmatically update permissions so the calling app can write to the created topic
+
+    # ideally we could check beforehand if the topic already exists instead of using exceptions as control flow
+    # but that is not in the AdminClient API
+    for topic, f in topic_futures.items():
+        try:
+            f.result()  # The result itself is None
+            logger.info(f"Topic {topic} created")
+        except KafkaException as ke:
+            if ke.args[0].code() == KafkaError.TOPIC_ALREADY_EXISTS:
+                logger.info(f"Topic {topic} already exists")
+            else:
+                raise
+
+
+def send_coupon_code_request_event_to_event_bus(event_name, event_properties):
+    """
+    Sends a coupon code request event to the event bus.
+    """
+    try:
+        event_producer = ProducerFactory.get_or_create_event_producer(
+            settings.COUPON_CODE_REQUEST_TOPIC_NAME,
+            StringSerializer('utf-8'),
+            CouponCodeRequestEventSerializer.get_serializer()
+        )
+        event_producer.produce(
+            settings.COUPON_CODE_REQUEST_TOPIC_NAME,
+            key=str(event_name),
+            value=CouponCodeRequestEvent(**event_properties),
+            on_delivery=verify_event
+        )
+        event_producer.poll()
+    except ValueSerializationError as vse:
+        logger.exception(vse)
+
+
+def verify_event(err, evt):
+    """
+    Simple callback method for debugging event production
+    :param err: Error if event production failed
+    :param evt: Event that was delivered
+    """
+    if err is not None:
+        logger.warning(f"Event delivery failed: {err}")
+    else:
+        # Don't log msg.value() because it may contain userids and/or emails
```

Comment:
This comment is now inaccurate since there are no emails in the new event and we don't actually care about userids since they are not PII